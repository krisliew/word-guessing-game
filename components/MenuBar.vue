<template>
<nav class="navbar navbar-expand-sm fixed-top justify-content-center">
  <ul class="navbar-nav">
    <!-- <li class="nav-item">
      <nuxt-link to="/play" :class="playClass">
            Play
        </nuxt-link>
    </li>
    <li class="nav-item">
      <nuxt-link to="/faq" :class="faqClass">
            Instruction
          </nuxt-link>
    </li> -->
    <li class="nav-item">
      <nuxt-link  v-if="activeSwap" to="/play" class="nav-link active">Play</nuxt-link>
      <nuxt-link  v-else to="/play" class="nav-link">Play</nuxt-link>
    </li>
    <li class="nav-item">
      <nuxt-link v-if="activeSwap" to="/faq" class="nav-link">Instruction</nuxt-link>
      <nuxt-link v-else to="/faq" class="nav-link active">Instruction</nuxt-link>
    </li>
  </ul>
    <!-- <nav class="navbar fixed" role="navigation" aria-label="main navigation">
      <div class="navbar-brand">
        <span class="navbar-item">
          <h2 class="title">Phrases</h2>
        </span>

        <a role="button" class="navbar-burger burger" aria-label="menu" aria-expanded="false" data-target="navbarBasicExample">
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
        </a>
      </div>

      <div id="navbarBasicExample" class="navbar-menu">
        <div class="navbar-start">
          <nuxt-link to="/" class="navbar-item">
            Home
          </nuxt-link>
          <nuxt-link to="/play" class="navbar-item">
            Play
          </nuxt-link>
          <nuxt-link to="/faq" class="navbar-item">
            Instruction 
          </nuxt-link>
        </div> -->

        <!-- <div class="navbar-end"> **** test
          <div class="navbar-item">
            <div class="buttons">
              <button v-show="!currentUser"
                      class="button is-primary"
                      @click="startSignUp">
                <strong>Sign up</strong>
              </button>
              <button v-show="!currentUser"
                      class="button is-light"
                      @click="startSignIn">
                Sign in
              </button>
              <button v-show="currentUser"
                      class="button is-primary"
                      :class="{'is-loading': sendingSignOut}"
                      @click="submitSignOut">
                <strong>Sign Out</strong>
              </button>
            </div>
          </div>
        </div>  
      </div> **** test -->
      <!-- <AuthModal :active="authActive"
                 :loading="authLoading"
                 :error-message="authError"
                 :title="authText"
                 :submit-text="authText"
                 @cancel="cancelAuth"
                 @submit="submitAuth"
      /> -->
    </nav>
</template>

<script>
/* eslint-disable */
// import AuthModal from "@/components/AuthModal";
import { mapActions, mapGetters } from "vuex";

export default {
  data(){
    return{}
  },
  created() {
    this.checkAccess();
  },
  computed:{
    activeSwap(){
      if(this.$route.path == "/play"){
        return true
      }else if(this.$route.path == "/faq"){
        return false;
      }      
    }
  },
  mounted(){
    // console.log("MenuBar.vue => Current Route: " + this.$route.path);
    // if(this.$route.path == "/play"){
    //   this.activeSwap = true;
    // }else if(this.$route.path == "/faq"){
    //   this.activeSwap = false;
    // }
  },
  methods: {
    ...mapActions("auth", [
      "startSignUp",
      "startSignIn",
      "endSignUp",
      "endSignIn",
      "submitSignUp",
      "submitSignIn",
      "submitSignOut",
      "checkAccess"
    ]),
    cancelAuth() {
      if (this.signingUp) {
        this.endSignUp();
      } else if (this.signingIn) {
        this.endSignIn();
      }
    },
    submitAuth(data) {
      if (this.signingUp) {
        this.submitSignUp(data);
      } else if (this.signingIn) {
        this.submitSignIn(data);
      }
    }
  }
/* ,components: { AuthModal },
  computed: {
    ...mapGetters("auth", [
      "signingUp",
      "signingIn",
      "signUpError",
      "signInError",
      "sendingSignUp",
      "sendingSignIn",
      "sendingSignOut",
      "currentUser"
    ]),
    authActive() {
      return this.signingUp || this.signingIn;
    },
    authLoading() {
      return this.sendingSignUp || this.sendingSignIn;
    },
    authError() {
      return this.signUpError || this.signInError;
    },
    authText() {
      if (this.signingUp) {
        return "Sign Up";
      } else if (this.signingIn) {
        return "Sign In";
      }

      return "";
    }
   }
*/
};
</script>

<style scoped>
nav {
  background-color: #00d1b2;
}

nav li a{
  color:white;
  font-size: 20px;
  font-family: 'Comic Sans MS';
}

nav .active{
  text-decoration: underline;
}

nav li a:hover, nav li:hover{
  background-color:white;
  color:black;
}
</style>
