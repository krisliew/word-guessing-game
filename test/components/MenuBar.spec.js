import MenuBar from "@/components/MenuBar.vue";
import { makeVueMock } from "@/test/utils";

describe("MenuBar", () => {
  const { wrapper } = makeVueMock(MenuBar, {
    shallow: true
  });

  test("is vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders regular", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test("no user - signing up", () => {
    const { wrapper, store } = makeVueMock(MenuBar, {
      shallow: true
    });
    store.commit("auth/setSigningUp", true);
    expect(wrapper.element).toMatchSnapshot();
  });

  test("no user - signing in", () => {
    const { wrapper, store } = makeVueMock(MenuBar, {
      shallow: true
    });
    store.commit("auth/setSigningIn", true);
    expect(wrapper.element).toMatchSnapshot();
  });

  test("has user", () => {
    const { wrapper, store } = makeVueMock(MenuBar, {
      shallow: true
    });
    store.commit("auth/setCurrentUser", { id: "123", username: "PandaBear" });
    expect(wrapper.element).toMatchSnapshot();
  });

  test("has user - signing out", () => {
    const { wrapper, store } = makeVueMock(MenuBar, {
      shallow: true
    });
    store.commit("auth/setCurrentUser", { id: "123", username: "PandaBear" });
    store.commit("auth/setSendingSignOut", true);
    expect(wrapper.element).toMatchSnapshot();
  });
});
