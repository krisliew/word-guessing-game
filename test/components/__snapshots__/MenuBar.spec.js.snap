// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar has user - signing out 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <span
      class="navbar-item"
    >
      <h2
        class="title"
      >
        Phrases
      </h2>
    </span>
     
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger burger"
      data-target="navbarBasicExample"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
    id="navbarBasicExample"
  >
    <div
      class="navbar-start"
    >
      <nuxt-link-stub
        class="navbar-item"
        to="/"
      >
        
        Home
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/play"
      >
        
        Play
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/create"
      >
        
        Create
      
      </nuxt-link-stub>
    </div>
     
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item"
      >
        <div
          class="buttons"
        >
          <button
            class="button is-primary"
            style="display: none;"
          >
            <strong>
              Sign up
            </strong>
          </button>
           
          <button
            class="button is-light"
            style="display: none;"
          >
            
            Sign in
          
          </button>
           
          <button
            class="button is-primary is-loading"
            style=""
          >
            <strong>
              Sign Out
            </strong>
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <authmodal-stub
    canceltext="Cancel"
    errormessage=""
    submittext=""
    title=""
  />
</nav>
`;

exports[`MenuBar has user 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <span
      class="navbar-item"
    >
      <h2
        class="title"
      >
        Phrases
      </h2>
    </span>
     
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger burger"
      data-target="navbarBasicExample"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
    id="navbarBasicExample"
  >
    <div
      class="navbar-start"
    >
      <nuxt-link-stub
        class="navbar-item"
        to="/"
      >
        
        Home
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/play"
      >
        
        Play
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/create"
      >
        
        Create
      
      </nuxt-link-stub>
    </div>
     
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item"
      >
        <div
          class="buttons"
        >
          <button
            class="button is-primary"
            style="display: none;"
          >
            <strong>
              Sign up
            </strong>
          </button>
           
          <button
            class="button is-light"
            style="display: none;"
          >
            
            Sign in
          
          </button>
           
          <button
            class="button is-primary"
            style=""
          >
            <strong>
              Sign Out
            </strong>
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <authmodal-stub
    canceltext="Cancel"
    errormessage=""
    submittext=""
    title=""
  />
</nav>
`;

exports[`MenuBar no user - signing in 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <span
      class="navbar-item"
    >
      <h2
        class="title"
      >
        Phrases
      </h2>
    </span>
     
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger burger"
      data-target="navbarBasicExample"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
    id="navbarBasicExample"
  >
    <div
      class="navbar-start"
    >
      <nuxt-link-stub
        class="navbar-item"
        to="/"
      >
        
        Home
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/play"
      >
        
        Play
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/create"
      >
        
        Create
      
      </nuxt-link-stub>
    </div>
     
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item"
      >
        <div
          class="buttons"
        >
          <button
            class="button is-primary"
          >
            <strong>
              Sign up
            </strong>
          </button>
           
          <button
            class="button is-light"
          >
            
            Sign in
          
          </button>
           
          <button
            class="button is-primary"
            style="display: none;"
          >
            <strong>
              Sign Out
            </strong>
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <authmodal-stub
    active="true"
    canceltext="Cancel"
    errormessage=""
    submittext="Sign In"
    title="Sign In"
  />
</nav>
`;

exports[`MenuBar no user - signing up 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <span
      class="navbar-item"
    >
      <h2
        class="title"
      >
        Phrases
      </h2>
    </span>
     
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger burger"
      data-target="navbarBasicExample"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
    id="navbarBasicExample"
  >
    <div
      class="navbar-start"
    >
      <nuxt-link-stub
        class="navbar-item"
        to="/"
      >
        
        Home
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/play"
      >
        
        Play
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/create"
      >
        
        Create
      
      </nuxt-link-stub>
    </div>
     
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item"
      >
        <div
          class="buttons"
        >
          <button
            class="button is-primary"
          >
            <strong>
              Sign up
            </strong>
          </button>
           
          <button
            class="button is-light"
          >
            
            Sign in
          
          </button>
           
          <button
            class="button is-primary"
            style="display: none;"
          >
            <strong>
              Sign Out
            </strong>
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <authmodal-stub
    active="true"
    canceltext="Cancel"
    errormessage=""
    submittext="Sign Up"
    title="Sign Up"
  />
</nav>
`;

exports[`MenuBar renders regular 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <span
      class="navbar-item"
    >
      <h2
        class="title"
      >
        Phrases
      </h2>
    </span>
     
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger burger"
      data-target="navbarBasicExample"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
    id="navbarBasicExample"
  >
    <div
      class="navbar-start"
    >
      <nuxt-link-stub
        class="navbar-item"
        to="/"
      >
        
        Home
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/play"
      >
        
        Play
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/create"
      >
        
        Create
      
      </nuxt-link-stub>
    </div>
     
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item"
      >
        <div
          class="buttons"
        >
          <button
            class="button is-primary"
          >
            <strong>
              Sign up
            </strong>
          </button>
           
          <button
            class="button is-light"
          >
            
            Sign in
          
          </button>
           
          <button
            class="button is-primary"
            style="display: none;"
          >
            <strong>
              Sign Out
            </strong>
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <authmodal-stub
    canceltext="Cancel"
    errormessage=""
    submittext=""
    title=""
  />
</nav>
`;
